### PR TITLE
Allowing config.signatureVersion to be used when signing requests

### PR DESCRIPTION
**Pull request by praf**
This change allows requests to a service to be signed if the api.json doesn't specify a signatureVersion or authtype, but the configuration has a signatureVersion specified

Prior to this change, the logic to create a signer would use signatureVersion specified in the config, but the code would never reach it if api.json signatureVersion or authtype weren't defined.

This required an update to test data setup in response.spec.js (since the requests are now being signed, but the signing was failing because the service name for the signer was ending up as undefined)

##### Checklist

- [x] `npm run test` passes

### CODE REVIEW

File: test/response.spec.js [hunk 1]
```diff
@@ -15,6 +15,9 @@
     beforeEach(function() {
       service = new AWS.Service({
         apiConfig: new AWS.Model.Api({
+          metadata: {
+            signingName: 'ocean-wave'
+          },
```
Comment (AllanFly120) [inline test/response.spec.js]: Why is this necessary?

Comment (praf) [reply test/response.spec.js]: I mention that in the commit body:
This required an update to test data setup in response.spec.js (since the requests are now being signed, but the signing was failing because the service name for the signer was ending up as undefined)

Basically, the service that was being setup by new AWS.service has config.signatureVersion set to "V4", therefore, these requests are now signed. However, signing requires a signingName to be set. I chose 'ocean-wave' as a test service name.

**Comment by AllanFly120** [comment]
Ok looks good to me. Can you add a change log for this commit? https://github.com/aws/aws-sdk-js/blob/master/CONTRIBUTING.md#changelog
